src/main/java/uk/gov/hmcts/reform/cwrdapi/service/JsrValidatorInitializer.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.reform.cwrdapi.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.PostConstruct;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+
+@Component
+@Slf4j
+public class JsrValidatorInitializer<T> {
+
+    private Validator validator;
+
+    private Set<ConstraintViolation<T>> constraintViolations;
+
+    @Value("${logging-component-name}")
+    private String logComponentName;
+
+    @PostConstruct
+    public void initializeFactory() {
+        ValidatorFactory factory = Validation.buildDefaultValidatorFactory();
+        validator = factory.getValidator();
+    }
+
+    /**
+     * JSR validation.
+     *
+     * @param domains List
+     * @return List binder list
+     */
+    public List<T> getInvalidJsrRecords(List<T> domains) {
+
+        log.info("{}:: JsrValidatorInitializer data processing validate starts::", logComponentName);
+        this.constraintViolations = new LinkedHashSet<>();
+        List<T> invalidList = new ArrayList<>();
+        domains.forEach(domain -> {
+            Set<ConstraintViolation<T>> constraintViolations = validator.validate(domain);
+            if (constraintViolations.size() > 0) {
+                invalidList.add(domain);
+            }
+            this.constraintViolations.addAll(constraintViolations);
```

Comment:
I think this code should be inside the if statement like `if (constraintViolations.size() > 0) {
                invalidList.add(domain);
this.constraintViolations.addAll(constraintViolations);
            }`
else I think empty constraintViolations will be added

Reply:
No...it is working :)